bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/nifs/NifsRule.java

```diff
@@ -191,29 +193,31 @@ public IP2Problems check(ObservationElements elements)  {
                 if (baseTarget == null || oiTarget.isEmpty() || aoTarget.isEmpty()) return null;
                 //now, let's compare coordinates. If they are the same, raise an error
 
-                double baseC1 = baseTarget.getTarget().getRaHours();
-                double baseC2 = baseTarget.getTarget().getDecDegrees();
+                final Option<Long> when = elements.getSchedulingBlock().map(SchedulingBlock::start);
 
-                double oiC1 = oiTarget.getValue().getTarget().getRaHours();
-                double oiC2 = oiTarget.getValue().getTarget().getDecDegrees();
+                baseTarget.getTarget().getRaHours(when).foreach(baseC1 ->
+                baseTarget.getTarget().getDecDegrees(when).foreach(baseC2 ->
 
-                double aoC1 = aoTarget.getValue().getTarget().getRaHours();
-                double aoC2 = aoTarget.getValue().getTarget().getDecDegrees();
+                oiTarget.getValue().getTarget().getRaHours(when).foreach(oiC1 ->
+                oiTarget.getValue().getTarget().getDecDegrees(when).foreach(oiC2 ->
 
-
-                if (Double.compare(baseC1, oiC1) == 0
-                        &&
-                        Double.compare(baseC2, oiC2) == 0
-                        &&
-                        Double.compare(oiC1, aoC1) == 0
-                        &&
-                        Double.compare(oiC2, aoC2) == 0) {
-                    P2Problems prob = new P2Problems();
-                    prob.addError(PREFIX + "OI_AO_SAME_AS_SCIENCE_TARGET_RULE", MESSAGE, elements.getTargetObsComponentNode().getValue());
-                    return prob;
+                aoTarget.getValue().getTarget().getRaHours(when).foreach(aoC1 ->
+                aoTarget.getValue().getTarget().getDecDegrees(when).foreach(aoC2 -> {
+                    if (Double.compare(baseC1, oiC1) == 0
```

Comment:
Usually targets aren't computed but rather typed in directly or more likely looked up in simbad or guide star catalogs.  Here we'll potentially be accounting for proper motion at some point but the same calculation will apply to all the targets and will compute the same values so nothing about how this check works is fundamentally changing.

You're right though and I imagine it would make sense to define some minimum angular separation that still counts as the same star if possible to do in all cases(?).  We'd need to hunt down all the places we do this kind of thing and use the same criterion everywhere as a separate issue.